ecr-login/api/factory.go

```diff
@@ -56,6 +57,16 @@ func (defaultClientFactory DefaultClientFactory) NewClientWithDefaults() Client
 	})
 }
 
+func (defaultClientFactory DefaultClientFactory) NewClientWithFipsEndpoint(endpoint, region string) Client {
```

Comment:
I think this function name doesn't make sense with what it does.  Right now, the name says "fips", but it really accepts any arbitrary name.  I think there are two possible options here:

1. Change the name to just be "NewClientWithEndpoint"
2. Move the endpoint-generation logic in here

I think I prefer option 2, since that makes the endpoint-generation logic usable by other consumers of this code.

Reply:
done